tests/test_polynomial.py

```diff
@@ -5,141 +5,90 @@
 
 class TestPolynomial(unittest.TestCase):
 
+    def setUp(self):
+        self.p1 = Polynomial(1, 2, 3)
+        self.p2 = Polynomial(3, 2, 1)
+        self.scalar = 5
+        self.zero_poly = Polynomial()
+
     def test_init(self):
-        # Test initialization from a sequence
-        p = Polynomial([1, 2, 3])
-        self.assertEqual(p.coeffs, [1, 2, 3])
+        # Test various ways of initializing a Polynomial
+        self.assertEqual(Polynomial(1, 2, 3).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial([1, 2, 3]).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial(self.p1).coeffs, self.p1.coeffs)
+        self.assertEqual(Polynomial(5).coeffs, [5])
+        self.assertEqual(Polynomial().coeffs, [])
+        self.assertEqual(Polynomial([1, 2, 3, 0, 0, 0]).coeffs, [1, 2, 3])
 
-        # Test initialization from scalars
-        p = Polynomial(1, 2, 3)
-        self.assertEqual(p.coeffs, [1, 2, 3])
+    def test_trim(self):
+        # Test the trimming of trailing zeros in a polynomial's coefficients
+        p = Polynomial([1, 0, 0, 0, 2])
+        p.trim()
+        self.assertEqual(p.coeffs, [1, 0, 0, 0, 2])
 
-        # Test copy constructor
-        p_copy = Polynomial(p)
-        self.assertEqual(p_copy.coeffs, p.coeffs)
+    def test_call(self):
+        # Test polynomial evaluation at given values
+        self.assertEqual(self.p1(0), 1)
+        self.assertEqual(self.p1(1), 6)
+        self.assertEqual(self.p1(2), 17)
 
-        # Test initialization with single scalar
-        p = Polynomial(5)
-        self.assertEqual(p.coeffs, [5])
+    def test_eq(self):
```

Comment:
Please don't unify the test cases. The unit tests should be as atomic as possible

Reply:
I think in the case of not very large tests, the chosen approach can be maintained